.github/workflows/ci.yml

```
@@ -291,73 +274,101 @@ jobs:
       - name: Install dependencies
         run: dart pub get
 
+      - uses: realm/ci-actions/mdb-realm/deployApps@63fb0c1678dd750e92709bb6c78aaafd03c5fade
+        with:
+          realmUrl: ${{ env.BAAS_URL }}
+          atlasUrl: ${{ secrets.ATLAS_QA_URL }}
+          projectId: ${{ env.BAAS_PROJECT_ID }}
+          apiKey: ${{ env.BAAS_API_KEY }}
+          privateApiKey: ${{ env.BAAS_PRIVATE_API_KEY }}
+          clusterName: ${{ env.BAAS_CLUSTER }}
+          useExistingCluster: true
+
       - name: Run tests
         run: ${{ matrix.architecture == 'arm' && 'arch -arm64 ' || '' }}dart test -r expanded -j 1 --test-randomize-ordering-seed random
 
 # Flutter jobs
 
-  flutter-desktop:
-    runs-on: ${{ matrix.platform == 'linux' && 'ubuntu' || matrix.platform }}-latest
+  flutter-desktop-tests:
     strategy:
       fail-fast: false
       matrix:
-        platform: [ 'macos', 'windows', 'linux']
-    name: Flutter ${{ matrix.platform }}
+        include:
+            - os: macos
+              app: fm
+            - os: windows
+              app: fw
+            - os: linux
+              app: fl
+    runs-on: ${{ matrix.os == 'linux' && 'ubuntu' || matrix.os }}-latest
+    name: Flutter tests on ${{matrix.os  }}
     env:
       BAAS_CLUSTER: ${{ needs.deploy-cluster.outputs.clusterName }}
-      BAAS_DIFFERENTIATOR: flutter-${{ matrix.platform }}
+      BAAS_DIFFERENTIATOR:  ${{ matrix.app }}${{ github.run_id }}${{ github.run_attempt }}
     needs:
-      - baas-matrix
       - deploy-cluster
       - build-native
+      - baas-matrix
     steps:
+
       - name: Checkout
         uses: actions/checkout@v2
         with:
           submodules: false
 
       - name: Setup GTK
-        if: ${{ matrix.platform == 'linux' }}
+        if: ${{ matrix.os == 'linux' }}
         run: |
           sudo apt-get update -y
           sudo apt-get install -y libgtk-3-dev xvfb
 
       - name: Setup Ninja
-        if: ${{ matrix.platform == 'linux' }}
+        if: ${{ matrix.os == 'linux' }}
         uses: seanmiddleditch/gha-setup-ninja@master
 
       - name: Fetch artifacts
         uses: actions/download-artifact@v2
         with:
-          name: librealm-${{ matrix.platform }}
-          path: binary/${{ matrix.platform }}
+          name: librealm-${{ matrix.os }}
+          path: binary/${{ matrix.os }}
 
       - name: Setup Flutter
         uses: subosito/flutter-action@v2
         with:
           channel: 'stable'
 
       - name: Enable Flutter Desktop support
-        run: flutter config --enable-${{ matrix.platform }}-desktop
+        run: flutter config --enable-${{ matrix.os }}-desktop
 
       - name: Install dependencies
         run: flutter pub get
 
+      - uses: realm/ci-actions/mdb-realm/deployApps@63fb0c1678dd750e92709bb6c78aaafd03c5fade
```

Might be a good idea to add a comment here and for the other jobs
```suggestion
      # This will be a no-op under normal circumstances since the cluster would have been deployed
      # in deploy-cluster. It is needed in case we want to re-run the job after the cluster has been reaped.
      - uses: realm/ci-actions/mdb-realm/deployApps@63fb0c1678dd750e92709bb6c78aaafd03c5fade
```